CHANGELOG.md

```diff
@@ -1,5 +1,57 @@
 # Change Log
 
+## [0.21.0](https://github.com/redhat-developer/vscode-xml/milestone/29?closed=1) (June 29, 2022)
+
+### Enhancements
+
+ * Add button to double heap size on OOM. See [#718](https://github.com/redhat-developer/vscode-xml/pull/718).
+ * CodeAction for cvc-elt.1.a. See [#599](https://github.com/redhat-developer/vscode-xml/issues/599).
+ * TextMate grammar for *.dtd files. See [#313](https://github.com/redhat-developer/vscode-xml/issues/313).
+ * CodeAction for cvc-complex-type.2.4.b. See [lemminx#1218](https://github.com/eclipse/lemminx/issues/1218).
+ * Support folding for <!DOCTYPE. See [lemminx#1213](https://github.com/eclipse/lemminx/issues/1213).
+ * Added folding setting to keep displaying the closing tag after folding. See [lemminx#1209](https://github.com/eclipse/lemminx/pull/1209).
+ * Provide basic experimental formatter which supports invalid XML. See [lemminx#1195](https://github.com/eclipse/lemminx/issues/1195).
+ * Format with xml:space. See [lemminx#826](https://github.com/eclipse/lemminx/issues/826).
+ * Code action for `SemicolonRequiredInReference`. See [lemminx#665](https://github.com/eclipse/lemminx/issues/665).
+ * Improve XML formatter (option to have Tags in the same line). See [lemminx#594](https://github.com/eclipse/lemminx/issues/594).
+
+### Performance
+
+ * Cancel process of code action. See [lemminx#1220](https://github.com/eclipse/lemminx/issues/1220).
+ * Parse of DOM document should be not done in a Thread. See [lemminx#1216](https://github.com/eclipse/lemminx/pull/1216).
+ * Improve DOM parser memory. See [lemminx#1211](https://github.com/eclipse/lemminx/pull/1211).
+ * Improve XML scanner memory. See [lemminx#1206](https://github.com/eclipse/lemminx/pull/1206).
+ * Wait a bit after change before sending diagnostics. See [lemminx#1162](https://github.com/eclipse/lemminx/issues/1162).
+ * Improve CodeAction performance with CodeAction#data & resolveCodeAction. See [lemminx#941](https://github.com/eclipse/lemminx/issues/941).
+ * Support for `completionItem/resolve`. See [lemminx#616](https://github.com/eclipse/lemminx/issues/616).
+
+### Bug Fixes
+
+ * externalXmlSettings.xmlFileAssociations not used in getXMLSettings. See [#698](https://github.com/redhat-developer/vscode-xml/issues/698).
+ * XSD with enumeration restriction with 1000+ entries fail type assist. See [#679](https://github.com/redhat-developer/vscode-xml/issues/679).
```

Comment:
This issue was resolved by the work for completionItem/resolve so it's a duplicate. Maybe worth keeping it out.